trunk/SUAVE/Analyses/Aerodynamics/AVL_Inviscid.py

```diff
@@ -0,0 +1,338 @@
+
+# Package imports
+import time
+import pylab as plt
+import os
+import numpy as np
+import sys
+import sklearn
+from sklearn import gaussian_process
+from sklearn import neighbors
+from sklearn import svm
+from shutil import rmtree
+from warnings import warn
+
+## ----------------------------------------------------------------------
+##  Class
+## ----------------------------------------------------------------------
+
+class AVL_Inviscid(Aerodynamics):
+    """ SUAVE.Analyses.Aerodynamics.AVL
+        aerodynamic model that performs a vortex lattice analysis using AVL
+        (Athena Vortex Lattice, by Mark Drela of MIT).
+
+        this class is callable, see self.__call__
+
+    """
+
+    def __defaults__(self):
+        self.tag        = 'avl'
+        self.keep_files = True
+
+        self.settings = Settings()
+
+        self.current_status = Data()
+        self.current_status.batch_index = 0
+        self.current_status.batch_file  = None
+        self.current_status.deck_file   = None
+        self.current_status.cases       = None
+        
+        self.geometry = None
+        
+        self.settings.filenames.log_filename = sys.stdout
+        self.settings.filenames.err_filename = sys.stderr
+        
+        # Conditions table, used for surrogate model training
+        self.training = Data()   
+        
+        #HALE UAV Cruise
+        self.training.angle_of_attack  = np.array([0 , 2, 4, 6 ,8, 10]) * Units.deg
+        self.training.Mach             = np.array([0.075, 0.1, 0.125 ,0.15,0.175, 0.2])       
+    
+        # Standard subsonic/transolic aircarft
+        #self.training.angle_of_attack  = np.array([-2.,3.,8.])
+        #self.training.Mach             = np.array([0.3,0.7,0.85]) 
+        
+        self.training.lift_coefficient = None
+        self.training.drag_coefficient = None
+        self.training_file             = None
+        
+        # Surrogate model
+        self.surrogates = Data()
+        self.surrogates.moment_coefficient = None
+
+    def initialize(self):
+
+        geometry = self.geometry
+        self.tag      = 'avl_analysis_of_{}'.format(geometry.tag)
+
+        run_folder = self.settings.filenames.run_folder
+        #if os.path.exists(run_folder):
+            #if self.keep_files:
+                #warn('deleting old avl run files',Warning)
+            #rmtree(run_folder)
+        #os.mkdir(run_folder)
+        
+        # Sample training data
+        self.sample_training()
+    
+        # Build surrogate
+        self.build_surrogate()
+
+
+        return
+
+    def evaluate(self,state,settings,geometry):
+
+        # Unpack
+        surrogates = self.surrogates        
+        conditions = state.conditions
+        
+        mach = conditions.freestream.mach_number
+        AoA  = conditions.aerodynamics.angle_of_attack
+        lift_model = surrogates.lift_coefficient
+        drag_model = surrogates.drag_coefficient
+        moment_model = surrogates.moment_coefficient
```

Comment:
Unless we change how the aerodynamics and stability analyses are structured, I don't think it will be. Moment data should be in the stability analysis.